### PR TITLE
Add EnableDefaultChannelLeaveJoinMessages config setting

### PR DESCRIPTION
**Pull request by csduarte**
#### Summary
Adds a config property `EnableDefaultChannelLeaveJoinMessages` that allows for leave/join messages to be created in the default channel. The default setting is true. When set to false a leave/join message will not be created in the DB for the default channel.

Extra Context: A major issue with our deployment is around how many people are on-board and off-boarded every day. There are too many message, town hall becomes useless. We can filter those messages out, but then the user must fetch many pages before getting an actual post. We need to stop the creation of those messages. We're open to a better solution.

@jarredwitt

### CODE REVIEW

**Comment by jasonblais** [comment]
Thanks!

@dmeza We have a ticket ([PLT-3755](https://mattermost.atlassian.net/browse/PLT-3755)) and a [WIP PR](https://github.com/mattermost/mattermost-server/pull/7557) to condense join/leave messages to a single post when appropriate. Wondering your thoughts on that approach?

**Comment by jasonblais** [comment]
Thanks @dmeza. I'll queue for our PM team to discuss and get back to you.

**Comment by dmeza** [comment]
@jasonblais rebased this to latest from master.

**Comment by jasonblais** [comment]
@dmeza Thank you. I was away for the last couple of weeks, I've added this to my review queue now.

**Comment by jasonblais** [comment]
(Update: Currently waiting to hear feedback from developers)

**Comment by dmeza** [comment]
@jasonblais sound good. I'll add the config setting to `app/diagnostics.go`.

**Comment by jwilander** [comment]
I'll be taking a look at this later today

**Comment by dmeza** [comment]
@jasonblais @jwilander rebased to latest and added config value `EnableDefaultChannelLeaveJoinMessages` to `diagnostics.go`

**Comment by jwilander** [comment]
Hi @dmeza, so I looked through the code for the older compliance feature and it does rely on these messages to tell when a user joined a team or not. I talked it over with @jasonblais and we're still OK with taking this but we'd like to update the config setting to be `Experimental EnableDefaultChannelLeaveJoinMessages` just to help make sure any customers relying on the old compliance feature make sure to read the documentation before using this setting

Would you be able to make that update? Code looks good so after that update I can get this merged

**Comment by dmeza** [comment]
@jwilander sounds good. I'll make the change to `Experimental EnableDefaultChannelLeaveJoinMessages` . I'll let you know when it's ready.

**Comment by dmeza** [comment]
@jwilander ready. Made the change to `ExperimentalEnableDefaultChannelLeaveJoinMessages`.

File: app/channel.go [hunk 1]
```diff
@@ -63,13 +63,15 @@ func (a *App) JoinDefaultChannels(teamId string, user *model.User, channelRole s
...
 		}
 
-		if requestor == nil {
-			if err := a.postJoinTeamMessage(user, townSquare); err != nil {
-				l4g.Error(utils.T("api.channel.post_user_add_remove_message_and_forget.error"), err)
-			}
-		} else {
-			if err := a.PostAddToTeamMessage(requestor, user, townSquare, ""); err != nil {
-				l4g.Error(utils.T("api.channel.post_user_add_remove_message_and_forget.error"), err)
+		if *a.Config().ServiceSettings.ExperimentalEnableDefaultChannelLeaveJoinMessages == true {
```
Comment (jwilander) [inline app/channel.go]: Minor but this can just be `if *a.Config().ServiceSettings.ExperimentalEnableDefaultChannelLeaveJoinMessages {`

**Pull request merged**